internal/metrics/daemon.go

```diff
@@ -0,0 +1,202 @@
+package metrics
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/prometheus/common/model"
+)
+
+const (
+	workloadMetric = "workloadName"
+	scrapeTimeout  = 15 * time.Second
+)
+
+type ScrappeCallback func(data []model.Vector, labels map[string]string)
+
+type TargetMetric struct {
+	workloadName string
+	interval     time.Duration
+	urls         []string
+	scrappers    []*HTTPScraper
+	ctx          context.Context
+	cancel       context.CancelFunc
+	forcetrigger chan bool
+	lock         sync.RWMutex
+}
+
+func NewTargetMetric(workloadName string, interval time.Duration, urls []string) *TargetMetric {
+	return &TargetMetric{
+		workloadName: workloadName,
+		interval:     interval,
+		urls:         urls,
+		forcetrigger: make(chan bool),
+	}
+}
+func (tg *TargetMetric) ForceEvent() {
+	tg.forcetrigger <- true
+}
+
+func (tg *TargetMetric) Start(cb ScrappeCallback) {
```

Comment:
Is the callback really needed? Can't the logic of scraping and adding to TSDB be a method on `TargetMetric`?

Reply:
yeah, it's reported to the daemon, so the callback or channel is needed.

Reply:
Can't the whole logic that relies on this callback be in `TargetMetric`?

Reply:
done, fixed.